src/MDMTimingCurve.h

```diff
@@ -0,0 +1,152 @@
+
+   A spring will treat the duration property of the traits as a suggestion and may choose to
+   ignore it altogether.
+   */
+  MDMTimingCurveTypeSpring,
+
+} NS_SWIFT_NAME(TimingCurveType);
+
+/**
+ A generalized representation of a traits curve.
+ */
+typedef struct MDMTimingCurve {
+  /**
+   The type defines how to interpret the data values.
+   */
+  MDMTimingCurveType type;
+
+  /**
+   The data values corresponding with this curve.
+   */
+  CGFloat data[4];
+} NS_SWIFT_NAME(TimingCurve) MDMTimingCurve;
+
+/**
+ Creates a bezier traits curve with the provided control points.
+
+ A cubic bezier has four control points in total. We assume that the first control point is 0, 0 and
+ the last control point is 1, 1. This method requires that you provide the second and third control
+ points.
+
+ See the documentation for CAMediaTimingFunction for more information.
+ */
+// clang-format off
+FOUNDATION_EXTERN
+MDMTimingCurve MDMTimingCurveMakeBezier(CGFloat p1x, CGFloat p1y, CGFloat p2x, CGFloat p2y)
+    NS_SWIFT_NAME(TimingCurve(bezierWithP1x:p1y:p2x:p2y:));
+// clang-format on
+
+// clang-format off
+FOUNDATION_EXTERN
+MDMTimingCurve MDMTimingCurveFromTimingFunction(CAMediaTimingFunction * _Nonnull traitsFunction)
+    NS_SWIFT_NAME(TimingCurve(fromTimingFunction:));
+// clang-format on
+
+/**
+ Creates a spring curve with the provided configuration.
+
+ Tension and friction map to Core Animation's stiffness and damping, respectively.
+
+ See the documentation for CASpringAnimation for more information.
+ */
+// clang-format off
+FOUNDATION_EXTERN
+MDMTimingCurve MDMTimingCurveMakeSpring(CGFloat mass, CGFloat tension, CGFloat friction)
+    NS_SWIFT_NAME(TimingCurve(springWithMass:tension:friction:));
+// clang-format on
+
+/**
+ Creates a spring curve with the provided configuration.
+
+ Tension and friction map to Core Animation's stiffness and damping, respectively.
+
+ See the documentation for CASpringAnimation for more information.
+ */
+// clang-format off
+FOUNDATION_EXTERN
+MDMTimingCurve MDMTimingCurveMakeSpringWithInitialVelocity(CGFloat mass,
+                                                           CGFloat tension,
+                                                           CGFloat friction,
+                                                           CGFloat initialVelocity)
+    NS_SWIFT_NAME(TimingCurve(springWithMass:tension:friction:initialVelocity:));
+// clang-format on
+
+/**
+ For cubic bezier curves, returns a reversed cubic bezier curve. For all other curve types, a copy
+ of the original curve is returned.
+ */
+// clang-format off
+FOUNDATION_EXTERN MDMTimingCurve MDMTimingCurveReversedBezier(MDMTimingCurve traitsCurve)
+    NS_SWIFT_NAME(TimingCurve(reversedFromTimingCurve:));
+// clang-format on
+
+/**
+ Named indices for the bezier traits curve's data array.
+ */
+typedef NS_ENUM(NSUInteger, MDMTimingCurveBezierDataIndex) {
+  MDMTimingCurveBezierDataIndexP1X,
```

Comment:
Please provide explicit numeric values for these and below. This seems especially valuable since they will function as indices into zero-based arrays.

Reply:
Done.